[Lau Jun Jie] iP

BlopApple opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use
 
All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

BlopApple committed:
Added Greet, Echo and Exit functionalities

BlopApple committed:
Added Add and List functionalities

BlopApple committed:
Added Marking and Unmarking functionalities

BlopApple committed:
Added ToDo, Deadline and Event Tasks

BlopApple committed:
Added Error Handling for invalid commands

BlopApple committed:
Added Delete functionality

BlopApple committed:
Text Ui Testing

BlopApple committed:
Add Save feature

BlopApple committed:
Add Date and Time processing for Tasks

BlopApple committed:
Merge branch 'branch-Level-7'
Add save feature to master

BlopApple committed:
Merge branch 'branch-Level-8'
Resolve merge conflict on branch-Level-8

BlopApple committed:
Fix minor merge conflicts in Level-7 and Level-8

BlopApple committed:
Refactor and Clean-up with OOP

BlopApple committed:
Add packages

KJunWei commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+public abstract class Task {
+    private final String taskName;
+    private boolean status; // True = Done, False = Not Done
```
could perhaps change status to isCompleted so the code may be more understandable!

KJunWei commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
```
Good job on ensuring the level of abstraction within the class is the same!

KJunWei commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,132 @@
+            return new EmptyTask();
+        }
+    }
+    public String list() {
```
could maybe add a verb to the method name e.g getList()?

KJunWei commented on src/main/java/duke/taskmanager/task/ToDoTask.java:
```diff
@@ -0,0 +1,34 @@
+import duke.taskmanager.exceptions.EmptyTaskException;
+
+public class ToDoTask extends Task {
+    private static final String taskName_TYPE = "T";
```
the name of the variable could possibly be TASKNAME_TYPE?

KJunWei commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,132 @@
+    public String mark(int itemNumber) {
+        if (itemNumber > 0 && itemNumber <= this.taskList.size()) {
+            if (this.taskList.get(itemNumber - 1).getStatus()) {
+                return "\tThe task is already marked you dummy.\n";
```
I love the humor!

KJunWei reviewed (commented):
Great job overall! Do remember to include Javadoc comments for the classes and methods!

BlopApple committed:
Add Gradle support

BlopApple committed:
Add unit testing for ChatBot and TaskManager

BlopApple committed:
Update gitignore and build.gradle files

BlopApple committed:
Add JavaDocs for all existing classes

BlopApple committed:
Add tweaks to comply with coding standards

BlopApple committed:
Add keyword finding functionality

BlopApple committed:
Merge branch 'branch-A-JavaDoc'

BlopApple committed:
Merge branch 'branch-A-CodingStandard'

BlopApple committed:
Merge branch 'branch-Level-9'

ngquyduc commented on src/main/java/duke/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,149 @@
+                    this.isRunning = false;
+                    break;
+                case "list":
+                    System.out.println(wrapMessage(taskManager.list()));
```
Not sure I like the wrapMessage() here. It might be a good idea to abstract all the System.out.println() methods with an UI class.

ngquyduc commented on src/main/java/duke/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,149 @@
+     *
+     * @param input string of the input provided by the user
+     */
+    public void processCommand(String input) {
```
Perhaps class Parser should be an alternative to process all the commands.

ngquyduc commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+import java.util.Scanner;
+
+public class TaskManager {
+    private static final String FILEPATH = "tasklist.txt";
```
```suggestion
    private static final String FILE_PATH = "tasklist.txt";
```

ngquyduc commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+    public String unmark(int itemNumber) {
+        if (itemNumber > 0 && itemNumber <= this.taskList.size()) {
+            if (!(this.taskList.get(itemNumber - 1).isCompleted())) {
+                return "\tThe task is still not done you idiot.\n";
```
I like the humour 👍

ngquyduc commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append("\tI have the matching tasks displayed below:\n");
+            for (Integer index : keywordList) {
+                stringBuilder.append("\t").append(index + 1).append(") ").append(this.taskList.get(index)).append("\n");
```
Perhaps you can use the method ```String.Format()``` for easier reading.

ngquyduc commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+            }
+            fileWriter.close();
+        } catch (IOException exception) {
+            System.out.println(exception);
```
```suggestion
            System.error.println(exception);
```

ngquyduc reviewed (commented):
Overall, the code is clean and easy to follow 👍 .
However, you can consider my suggestions for improvements.

NicholasPatrick commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+
+public class TaskManager {
+    private static final String FILEPATH = "tasklist.txt";
+    private static final String DATE_FORMAT = "dd/MM/yyyy,HHmm";
```
Perhaps this constant belongs in its own class.

NicholasPatrick commented on src/main/java/duke/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,149 @@
+package duke.chatbot;
+
+import duke.taskmanager.TaskManager;
+import duke.taskmanager.task.*;
+import duke.taskmanager.exceptions.*;
```
Consider not importing wildcards.

NicholasPatrick commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+package duke.taskmanager;
+
+import duke.taskmanager.task.*;
```
Consider listing all the classes needed.

NicholasPatrick commented on src/main/java/duke/taskmanager/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+     * @param itemNumber index of the task to be marked as completed
+     * @return response message indicating the marking of the task
+     */
+    public String mark(int itemNumber) {
```
Consider making this method name more descriptive to show that it's not a getter. The same can be said for the `unmark` method.

BlopApple committed:
Fix minor code quality issues

BlopApple committed:
Add CheckStyle

BlopApple committed:
Add Basic GUI with no user interaction

BlopApple committed:
Add Basic GUI with user interaction

BlopApple committed:
Refactor GUI elements

BlopApple committed:
Merge branch 'branch-A-CheckStyle'

BlopApple committed:
Merge branch 'branch-Level-10'

BlopApple committed:
Add proper padding and sizes of GUI elements

Padding is added to InputContainer and DialogContainer
for a cleaner look.

Sizes of GUI elements that used to be magic literals were
properly abstracted out as class attributes of the various
GUI elements.

Font Size of text in DialogBoxes is fixed at 12.

Response messages were also updated to no longer include
tabspaces at the beginning for standardization of response formats.

BlopApple committed:
Add assertions

BlopApple committed:
Refactor code and improve code quality

BlopApple committed:
Add Streams

BlopApple committed:
Merge pull request #1 from BlopApple/branch-A-Assertions

Add assertions

BlopApple committed:
Merge pull request #2 from BlopApple/branch-A-Streams

Add Streams

BlopApple committed:
Merge branch 'master' into branch-A-CodeQuality

BlopApple committed:
Merge pull request #3 from BlopApple/branch-A-CodeQuality

Refactor code and improve code quality

BlopApple committed:
gradle.yml

BlopApple committed:
Rename main.yml to gradle.yml

BlopApple committed:
Add Update functionality for task

BlopApple committed:
Merge branch 'branch-C-Update'

BlopApple committed:
Add Command Handlers and implement Bye command

BlopApple committed:
Implement ListTask Command

BlopApple committed:
Implement adding of ToDo, Deadline and Event Task

Implement the TodoTaskCommandHandler, DeadlineTaskCommandHandler and
EventTaskCommandHanlder to add new task to the task list.

Temporarily suppress updateTask functionality.

BlopApple committed:
Implement UpdateTask command

BlopApple committed:
Update JavaDocs for implemented CommandHandlers

Implemented CommandHandlers include CommandHandlers for the
Bye, List, TodoTask, DeadlineTask, EventTask and UpdateTask Commands.

BlopApple committed:
Implement Mark, Unmark and Delete Task commands

BlopApple committed:
Implement Find Task command

BlopApple committed:
Add Command Manager

BlopApple committed:
Add more JUnit Test for Commmand Manager

BlopApple committed:
Improve Basic GUI elements

Include a dialog box background for dialog boxes.

Change display picture into a display bubble look.

Include proper padding and spacing for dialog box and display picture.

Change alignment of display picture to flush to the bottom.

BlopApple committed:
Add Logger to access previous commands for better UX

BlopApple committed:
Make window resizable

BlopApple committed:
Add styling to GUI elements

BlopApple committed:
Merge branch 'branch-CommandAbstraction'

Fully abstract commands from chatbot and taskmanager.

Implement command manager for chatbot to handle commands instead.

Previously command exceptions were handled by both chatbot and
taskmanager. The improved abstraction lets new command objects to handle
exceptions and all response messages.

BlopApple committed:
Merge branch 'branch-A-BetterGui'

Improve current GUI with more styling and functional elements.

BlopApple committed:
Fix minor conflict remnants

BlopApple committed:
Update build.gradle to include SnakeYAML

BlopApple committed:
Add Christina.yml to resources

BlopApple committed:
Update Command Handlers to use Personality class

BlopApple committed:
Update Exceptions to use Personality class

BlopApple committed:
Merge branch 'branch-A-Personality'

BlopApple committed:
Update personality yaml and name of save file

BlopApple committed:
Add annoyed display picture for chatbot

BlopApple committed:
Draft a User Guide in the README

BlopApple committed:
Update README.md

BlopApple committed:
Finish User Guide

BlopApple committed:
Update README.md

BlopApple committed:
Update README.md to include Acknowledgements

BlopApple committed:
Update README.md and fix bug in Find Command

BlopApple committed:
Improve code quality in command manager

BlopApple committed:
Add greeting with list size when save file is not empty
